src/main/java/com/cloudbees/jenkins/plugins/bitbucket/api/credentials/BitbucketOAuthAuthenticator.java

```diff
@@ -5,45 +5,60 @@
 import com.cloudbees.plugins.credentials.common.StandardUsernameCredentials;
 import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
 import com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl;
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.httpclient.HttpClient;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.model.OAuthConstants;
+import com.github.scribejava.core.oauth.OAuth20Service;
+import com.github.scribejava.httpclient.apache.ApacheHttpClientConfig;
+import com.github.scribejava.httpclient.apache.ApacheProvider;
 import hudson.model.Descriptor.FormException;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import jenkins.authentication.tokens.api.AuthenticationTokenException;
 import org.apache.commons.lang.StringUtils;
 import org.apache.http.HttpRequest;
-import org.scribe.model.OAuthConfig;
-import org.scribe.model.OAuthConstants;
-import org.scribe.model.Token;
 
 public class BitbucketOAuthAuthenticator extends BitbucketAuthenticator {
 
-    private Token token;
+    private OAuth2AccessToken token;
 
     /**
      * Constructor.
      *
      * @param credentials the key/pass that will be used
+     * @throws AuthenticationTokenException
      */
-    public BitbucketOAuthAuthenticator(StandardUsernamePasswordCredentials credentials) {
+    public BitbucketOAuthAuthenticator(StandardUsernamePasswordCredentials credentials) throws AuthenticationTokenException {
         super(credentials);
 
-        OAuthConfig config = new OAuthConfig(credentials.getUsername(), credentials.getPassword().getPlainText());
+        HttpClient httpClient = new ApacheProvider().createClient(ApacheHttpClientConfig.defaultConfig());
+        OAuth20Service service = new ServiceBuilder(credentials.getUsername())
+            .apiSecret(credentials.getPassword().getPlainText())
+            .httpClient(httpClient)
+         // .httpClientConfig(ApacheHttpClientConfig.defaultConfig()) the ServiceLoader does not work well with Jenkins plugin classloader
```

Comment:
Very nice PR! FTR core's `SetContextClassLoader` might help in situations like this.

Reply:
What means FTR ?

Reply:
> Very nice PR! FTR core's `SetContextClassLoader` might help in situations like this.

It works. Update master using SetContextClassLoader